lib/gag.py

```diff
@@ -0,0 +1,33 @@
+import datetime
+import mongoengine
+
+class Gag(mongoengine.Document):
+    """Gag details are stored here"""
+    content_type = mongoengine.StringField(required=True)
+    url = mongoengine.StringField(required=True)
+    created_at = mongoengine.DateTimeField(default=datetime.datetime.now)
+    tags = mongoengine.StringField()
+
+    def save_gags(gags):
+        for gag in gags:
+            existing_gag = Gag.objects(url=gag['url']).first()
+            if existing_gag == None:
+                tags = gag['tags']
+                gag_tags = []
+                for tag in tags:
+                    gag_tags.append(tag['key'])
+                newGag = Gag(content_type=gag['type'], url=gag['url'], tags=', '.join(gag_tags))
+                newGag.save()
+
+    def get_gags(type, count):
+        input_mapping = dict({'IMAGE': 'Photo', 'GIF': 'Animated', 'VIDEO':'EmbedVideo'})
```

Comment:
Overall X is not handled here. For reference, https://github.com/browserstack/tech-onboarding/tree/master/scraper

Reply:
so anything apart from 'IMAGE', 'GIF', 'VIDEO' will be considered as OVERALL and it will work. Let me know if I should implement is specifically.

Reply:
Anything apart from three will take any outliers which is not a good practice. Implement overall specifically. As anyone who reads the instruction will try running that

Reply:
Updated the code accordingly